tests/AutoloaderTest.php

```diff
@@ -154,7 +151,66 @@ public function testRegisterAutoloadTwiceDoesNotWork(): void
         }
     }
 
-    public function testCallLoadClassTwiceShouldWork(): void
+
+    /**
+     * @dataProvider getClassIndexData
+     */
+    public function testCallLoadClassTwiceShouldWork(array $classIndex, array $files): void
+    {
+        $this->createTestFiles($files);
+
+        $autoload = new Autoloader(self::ROOT_DIR);
+        $autoload->setClassIndex($classIndex);
+        $autoload->register(); // register autoload once.
+
+        $autoload->load('BarTest');
+        $autoload->load('BarTest');
+        $autoload->load('Logger'); // Logger is an alias to PrestaShopLogger
+        $autoload->load('Logger');
+        self::assertTrue(true);
+
+        spl_autoload_unregister([$autoload, 'load']);
+    }
+
+
+
+    /**
+     * @dataProvider getClassIndexData
+     */
+    public function testAutoloadCallsStaticMethods(array $classIndex, array $files): void
+    {
+        $this->createTestFiles($files);
+
+        $autoload = new Autoloader(self::ROOT_DIR);
+        $autoload->setClassIndex($classIndex);
+
+        spl_autoload_register([$autoload, 'load']);
+        \Baz::foo(); // @phpstan-ignore-line
+        spl_autoload_unregister([$autoload, 'load']);
+
+        self::assertSame('Hello world', \Baz::foo()); // @phpstan-ignore-line
+    }
+
+    /**
+     * @dataProvider getClassIndexData
+     */
+    public function testAutoloadDoesNotLoadIndexByDefault(array $classIndex, array $files): void
```

Comment:
This function name does not exactly match what is tested? It does part of your logic